aom/src/main/java/com/nedap/archie/aom/primitives/CTerminologyCode.java

```diff
@@ -94,14 +91,22 @@ public boolean isValidValue(TerminologyCode value) {
             return true;
         }
         if(isConstraintRequired()) {
-            if(value != null && value.getTerminologyId() != null && !value.getTerminologyId().equalsIgnoreCase("local") && !AOMUtils.isValueSetCode(value.getTerminologyId())) {
-                //this is a non-local terminology. If a term binding is there, we may be able to validate, if external, we wil not be able to
-                //so return true for now for non-local terminology values
+            if (value == null) return false;
+
+            List<String> values;
+            String terminologyId = value.getTerminologyId();
+            if (terminologyId == null || terminologyId.equalsIgnoreCase("local") || AOMUtils.isValueSetCode(value.getTerminologyId())) {
+                values = this.getValueSetExpanded();
+            } else if (terminologyId.equalsIgnoreCase("openehr")) {
```

Comment:
We started with openehr because we needed that at the moment and it is the only one which has a URL-to-code rule defined in Archie. We weren't aware of the templates in the ADL1.4 config yet.

Our suggestion would be adding support for openehr now and extend it with configurable templates later on when needed.

Reply:
Ok, sounds good!